src/user/user.service.ts

```diff
@@ -141,16 +134,20 @@ export class UserService {
     }
   }
 
-  async updateUserProfile(id: string, updateUserDto: UpdateUserDto, user: User) {
+  async updateUserProfile(id: string, updateUserDto: UpdateUserDto, authUser: User) {
     try {
-      if (!user) {
-        throw new NotFoundException("User doesn't exist");
+      const userProfile = await this.prismaService.userProfile.findFirst({
+        where: { id },
+      });
+
+      if (!userProfile) {
+        throw new NotFoundException(`User Profile with ID ${id} not found`);
       }
 
-      await this.validateUserAutherization(id, user.id);
+      await this.validateUserAutherization(userProfile.userId, authUser);
```

Comment:
are they different id and userProfile.userId?

Reply:
if they are the same  await this.validateUserAutherization(id, user.id) should be invoked before findFirst

Reply:
they could be different, authUser in validateUserAutherization(userProfile.userId, authUser) is user from token which might be anyone (owner, admin, or someone else) and we cannot invoke validateUserAutherization(userProfile.userId, authUser) before findFirst since we don't have owner id(userProfile.userId)